### PR TITLE
ref(perf): Move performance utility to its own folder

### PR DESCRIPTION
**Pull request by Zylphrex**
The number of utility functions for the performance product has steadily been
increasing. This moves many of them into their own dedicated folder at
`app/utils/performance` to make for better organization.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/performance/transactionVitals/types.tsx [hunk 1]
```diff
@@ -1,27 +1,3 @@
-import {ColumnType, WebVital} from 'app/utils/discover/fields';
-
-export type HistogramData = {
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/transactionVitals/types.tsx:3]: Turns out this was duplicated, so let's consolidate them into one.

File: src/sentry/static/sentry/app/utils/performance/histogram/types.ts [hunk 2]
```diff
@@ -4,3 +4,5 @@ type HistogramDataUnit = {
 };
 
 export type HistogramData = HistogramDataUnit[];
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/utils/performance/histogram/types.ts:6]: Slightly different from the original type to save a little typing and matches the naming of the type of an ECharts series a little better.

File: src/sentry/static/sentry/app/utils/performance/vitals/constants.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+export const WEB_VITAL_DETAILS: Record<WebVital, Vital> = {
+  [WebVital.FP]: {
+    slug: 'fp',
+    name: t('First Paint'),
+    acronym: 'FP',
+    description: t(
+      'Render time of the first pixel loaded in the viewport (may overlap with FCP).'
+    ),
+    poorThreshold: 3000,
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/utils/performance/vitals/constants.ts:13]: Renamed from `failureThreshold` to `poorThreshold`. A TODO item here is to consolidate the duplicate thresholds defined in vital details.

**Review by markstory — APPROVED** [review]
:clap:

**Review by dashed — APPROVED** [review]

**Pull request merged**